django/utils/text.py

```diff
@@ -11,7 +11,11 @@
 @keep_lazy_text
 def capfirst(x):
     """Capitalize the first letter of a string."""
-    return x and str(x)[0].upper() + str(x)[1:]
+    if x is None:
+        return None
+    else:
+        x = str(x)  # evaluate lazy strings (just once)
+        return x and x[0].upper() + x[1:]
```

Comment:
My comment: this does not exactly preserve behaviour in all cases involving false-y values. For example, `capfirst(0)` used to return `0`, now it returns `'0'`. Which is correct behaviour is not defined, and I would consider the behaviour of using `capfirst` on something that is not a string is not compatibility we need to maintain, especially given we haven't even documented `django.utils.text.capfirst` AFAICS. (The `capfirst` filter implementation is independent)